Sources/Core/OpaquePointer.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2022 Max Desiatov <max@desiatov.com>.
+// All Rights Reserved.
+// SPDX-License-Identifier: BSD-3
+
+@frozen
+public struct OpaquePointer {
+  @usableFromInline
+  internal var _rawValue: Builtin.RawPointer
+
+  @usableFromInline @_transparent
+  internal init(_ v: Builtin.RawPointer) {
+    _rawValue = v
+  }
+
+  @_transparent
+  public init?(bitPattern: Int) {
+    if bitPattern == 0 { return nil }
+    _rawValue = Builtin.inttoptr_Int32(bitPattern._value)
```

Comment:
This will truncate no?

Reply:
It will, but I had to use `Int32` here, as you've requested `Word` changes to `UInt` and `Int` to be removed from this PR in your comment above.

Reply:
I'm trying to get you to be more methodical about the changes.  Clearly there is a bug in `Int` and `UInt`, that should be fixed first.  Once that is fixed, adding an implementation here is simple.